pkg/mcs/registry/registry.go

```diff
@@ -72,13 +64,7 @@ func (r *ServiceRegistry) InstallAllGRPCServices(srv *server.Server, g *grpc.Ser
 // InstallAllRESTHandler installs all registered REST services.
 func (r *ServiceRegistry) InstallAllRESTHandler(srv *server.Server, h map[string]http.Handler) {
```

Comment:
How about turn `RegisterService` func to pointer receivers
`func (r *ServiceRegistry) RegisterService`

Reply:
Maybe we could file another PR to address this. Let's keep this PR focus on the failover.